Adds information about canary files and the quarantine process for malicious files (#3398)
elastic/security-docs: docs/getting-started/configure-integration-policy.asciidoc

extending the list of processes that {elastic-defend} considers malicious. Use the **Blocklist enabled** toggle
to enable or disable this feature for all hosts associated with the integration policy. To configure the blocklist, refer to <<blocklist>>.

> If *Detect* malware is enabled, when a malicious file is found it gets quarantined. Specifically {agent} removed from it's current location, obfuscates it (XOR encrypts it with the encryption key `ELASTIC`), moves it to a different folder, and renames it to `xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`. {agent} versions 8.5 and later moves quarantined files to the folder `[DriveLetter:]\.quarantine`, unless they are from the `C:` drive, in which case it moves them to `C:\Program Files\Elastic\Endpoint\state\.equarantine`. {agent} versions earlier than 8.5 move quarantined files to `[DriveLetter:]\.quarantine` regardless of which drive they were originally on.

Reviewer: Shouldn't that be `If *Prevent* malware is enabled` or possibly `If malware is enabled and in *Prevent* mode`...
Because only the Prevent mode would move to the quarantine folder.

> If *Detect* malware is enabled, when a malicious file is found it gets quarantined. Specifically {agent} removed from it's current location, obfuscates it (XOR encrypts it with the encryption key `ELASTIC`), moves it to a different folder, and renames it to `xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`. {agent} versions 8.5 and later moves quarantined files to the folder `[DriveLetter:]\.quarantine`, unless they are from the `C:` drive, in which case it moves them to `C:\Program Files\Elastic\Endpoint\state\.equarantine`. {agent} versions earlier than 8.5 move quarantined files to `[DriveLetter:]\.quarantine` regardless of which drive they were originally on.

Reviewer: Furthermore is the encryption key `ELASTIC` or `elastic` . I mean what exact case sensitivity is true here?

Author: confirmed it's ELASTIC

> If *Detect* malware is enabled, when a malicious file is found it gets quarantined. Specifically {agent} removed from it's current location, obfuscates it (XOR encrypts it with the encryption key `ELASTIC`), moves it to a different folder, and renames it to `xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`. {agent} versions 8.5 and later moves quarantined files to the folder `[DriveLetter:]\.quarantine`, unless they are from the `C:` drive, in which case it moves them to `C:\Program Files\Elastic\Endpoint\state\.equarantine`. {agent} versions earlier than 8.5 move quarantined files to `[DriveLetter:]\.quarantine` regardless of which drive they were originally on.

Author: ```suggestion
When *Detect* malware is enabled, {agent} will quarantine any malicious file it finds. Specifically {agent} will remove it from it's current location, obfuscate it (XOR encrypt it with the encryption key `ELASTIC`), move it to a different folder, and rename it to `xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`. {agent} versions 8.5 and later move quarantined files to the folder `[DriveLetter:]\.quarantine`, unless they are from the `C:` drive, in which case it moves them to `C:\Program Files\Elastic\Endpoint\state\.equarantine`. {agent} versions earlier than 8.5 move quarantined files to `[DriveLetter:]\.quarantine` regardless of which drive they were originally on.
```

> When *Detect* malware is enabled, {agent} will quarantine any malicious file it finds. Specifically {agent} will remove it from it's current location, obfuscate it (XOR encrypt it with the encryption key `ELASTIC`), move it to a different folder, and rename it to `xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`. {agent} versions 8.5 and later move quarantined files to the folder `[DriveLetter:]\.quarantine`, unless they are from the `C:` drive, in which case it moves them to `C:\Program Files\Elastic\Endpoint\state\.equarantine`. {agent} versions earlier than 8.5 move quarantined files to `[DriveLetter:]\.quarantine` regardless of which drive they were originally on.

Author: ```suggestion
When *Detect* malware is enabled, {agent} will quarantine any malicious file it finds. Specifically {agent} will remove it from it's current location, obfuscate it (XOR encrypt it with the encryption key `ELASTIC`), move it to a different folder, and rename it as a string like `318e70c2-af9b-4c3a-939d-11410b9a112c`. {agent} versions 8.5 and later move quarantined files to the folder `[DriveLetter:]\.quarantine`, unless they are from the `C:` drive, in which case it moves them to `C:\Program Files\Elastic\Endpoint\state\.equarantine`. {agent} versions earlier than 8.5 move quarantined files to `[DriveLetter:]\.quarantine` regardless of which drive they were originally on.
```

extending the list of processes that {elastic-defend} considers malicious. Use the **Blocklist enabled** toggle
to enable or disable this feature for all hosts associated with the integration policy. To configure the blocklist, refer to <<blocklist>>.

When *Detect* malware is enabled, {agent} will quarantine any malicious file it finds. Specifically {agent} will remove it from it's current location, obfuscate it (XOR encrypt it with the encryption key `ELASTIC`), move it to a different folder, and rename it as a string like `318e70c2-af9b-4c3a-939d-11410b9a112c`.

> - On MacOS, all {agent} versions move quarantined files to the folder `/System/Volumes/Data/.equarantine`.

Reviewer: Are you sure the MacOS directory is `/System/Volumes/Data/.equarantine` ? It seems to me looking to my installation instance that it is rather directory `/Library/Endgame/.equarantine`

Author: @roxana-gheorghe Please confirm

Reviewer 2: The path containing "Endgame" comes from an Endgame sensor, the path for Elastic should be /System/Volumes/Data/.equarantine.

- On Linux, all {agent} versions move quarantined files to the folder `.equarantine` at the root of the mount point of the file being quarantined.

> - On Windows, {agent} versions 8.5 and later move quarantined files to the folder `[DriveLetter:]\.quarantine`, unless they are from the `C:` drive, in which case it moves them to `C:\Program Files\Elastic\Endpoint\state\.equarantine`. {agent} versions earlier than 8.5 move quarantined files to `[DriveLetter:]\.quarantine` regardless of which drive they were originally on.

Reviewer 3: This last bullet is pretty dense, so adding the line break might improve readability. Your call!

```suggestion
- On Windows, {agent} versions 8.5 and later move quarantined files to the folder `[DriveLetter:]\.quarantine`, unless they are from the `C:` drive. These files are moved to `C:\Program Files\Elastic\Endpoint\state\.equarantine`. 
+
{agent} versions earlier than 8.5 move quarantined files to `[DriveLetter:]\.quarantine`, regardless of their original drive.
```

> To restore a quarantined file to its original state, including its location, you can <<add-exceptions, create an exception>> for the rule that identified it as malicious. If the exception would have prevented the file from being identified as malicious, {agent} will restore the file.

Reviewer 3: Left some minor suggestions for accuracy.  

```suggestion
To restore a quarantined file to its original state and location, <<add-exceptions, add an exception>> to the rule that identified the file as malicious. If the exception would've stopped the rule from identifying the file as malicious, {agent} restores the file. 
```

extending the list of processes that {elastic-defend} considers malicious. Use the **Blocklist enabled** toggle
to enable or disable this feature for all hosts associated with the integration policy. To configure the blocklist, refer to <<blocklist>>.

> When *Detect* malware is enabled, {agent} will quarantine any malicious file it finds. Specifically {agent} will remove the file from its current location, obfuscate it (XOR encrypt it with the encryption key `ELASTIC`), move it to a different folder, and rename it as a string like `318e70c2-af9b-4c3a-939d-11410b9a112c`.

Reviewer 4: A few different edits here, to streamline extra phrasing and reduce jargon (I don't know what XOR encryption is, but is mentioning it relevant here?):
```suggestion
When *Detect* is enabled for malware protection, {agent} will quarantine any malicious file it finds. Specifically {agent} will remove the file from its current location, encrypt it with the encryption key `ELASTIC`, move it to a different folder, and rename it as an ID string, such as `318e70c2-af9b-4c3a-939d-11410b9a112c`.
```

> - On macOS, all {agent} versions move quarantined files to the folder `/System/Volumes/Data/.equarantine`.
> - On Linux, all {agent} versions move quarantined files to the folder `.equarantine` at the root of the mount point of the file being quarantined.
> - On Windows, {agent} versions 8.5 and later move quarantined files to the folder `[DriveLetter:]\.quarantine`, unless they are from the `C:` drive. These files are moved to `C:\Program Files\Elastic\Endpoint\state\.equarantine`. 
> +
> {agent} versions earlier than 8.5 move quarantined files to `[DriveLetter:]\.quarantine`, regardless of their original drive.

Reviewer 4: A lot of repetition here, mostly of phrasing that feels like filler. A table might be cleaner, or reformat the list to focus on essentials (you might still need to add a lead-in for a list like this):
```suggestion
- macOS: `/System/Volumes/Data/.equarantine`
- Linux: `.equarantine` at the root of the mount point of the file being quarantined
- Windows - {agent} versions 8.5 and later: `[DriveLetter:]\.quarantine`, unless the files are from the `C:` drive. These files are moved to `C:\Program Files\Elastic\Endpoint\state\.equarantine`.
- Windows - {agent} versions 8.4 and earlier: `[DriveLetter:]\.quarantine`, regardless of the files' original drive
```

extending the list of processes that {elastic-defend} considers malicious. Use the **Blocklist enabled** toggle
to enable or disable this feature for all hosts associated with the integration policy. To configure the blocklist, refer to <<blocklist>>.

> 
> When *Detect* is enabled for malware protection, {agent} will quarantine any malicious file it finds. Specifically {agent} will remove the file from its current location, encrypt it with the encryption key `ELASTIC`, move it to a different folder, and rename it as an ID string, such as `318e70c2-af9b-4c3a-939d-11410b9a112c`.

Author: ```suggestion
```

> The quarantine folder location depends varies by operating system:

Author: ```suggestion
The quarantine folder location varies by operating system:
```

- macOS: `/System/Volumes/Data/.equarantine`
- Linux: `.equarantine` at the root of the mount point of the file being quarantined
- Windows - {agent} versions 8.5 and later: `[DriveLetter:]\.quarantine`, unless the files are from the `C:` drive. These files are moved to `C:\Program Files\Elastic\Endpoint\state\.equarantine`.

> - Windows - {agent} versions 8.4 and earlier: `[DriveLetter:]\.quarantine`, regardless of the files' original drive

Author: ```suggestion
- Windows - {agent} versions 8.4 and earlier: `[DriveLetter:]\.quarantine`, for any drive
```

Ransomware protection is a paid feature and is enabled by default if you have a https://www.elastic.co/pricing[Platinum or Enterprise license].
If you upgrade to a Platinum or Enterprise license from Basic or Gold, ransomware protection will be disabled by default.

> When ransomware protection is enabled, canary files are placed in targeted locations throughout the Windows file system. They provide an early warning system for potential ransomware activity. When canary files are modified, Elastic Defend will immediately generate a ransomware alert. If *prevent* ransomware is active, {elastic-defend} will terminate the process responsible for modifying the file.

Reviewer 3: Some suggestions to link the first and second sentences and address some of the passive voice.
  
```suggestion
When ransomware protection is enabled, canary files are placed in targeted locations throughout the Windows file system as an early warning system for potential ransomware activity. When canary files are modified, Elastic Defend immediately generates a ransomware alert. If *prevent* ransomware is active, {elastic-defend} terminates the process that modified the file.
```

Reviewer: For me is not clear what `targeted locations` means. Is there a specific directory in mind where we locate the canary files? I might be good to document this directory for Windows? And we might point out a bit more clear that canary files are not available on MacOS and Linux.

Author: @roxana-gheorghe Can you provide any insight into Andre's question about the location of canary files?

Reviewer 3: Do we want to publicize where we're placing the canary files? Would that make it easier for adversaries to evade the warning system we're putting in pace? I'm not trying to be contrary here. I'm just wondering if it's safer to obscure these details in the public docs.

Reviewer 4: +1 to @nastasha-solomon's edits to reduce passive voice, but that first sentence is pretty long now. Is "throughout the Windows file system" necessary? or maybe shorten to "on the host"?

On publicizing the location: this doesn't seem to identify specific locations, just that the files are out there somewhere.

Ransomware protection is a paid feature and is enabled by default if you have a https://www.elastic.co/pricing[Platinum or Enterprise license].
If you upgrade to a Platinum or Enterprise license from Basic or Gold, ransomware protection will be disabled by default.

> 
> When ransomware protection is enabled, canary files placed in targeted locations on your hosts provide an early warning system for potential ransomware activity. When a canary file is modified, Elastic Defend immediately generates a ransomware alert. If *prevent* ransomware is active, {elastic-defend} terminates the process that modified the file.

Author: ```suggestion
```

> When ransomware protection is enabled, canary files are placed in targeted locations throughout the Windows file system. They provide an early warning system for potential ransomware activity. When canary files are modified, Elastic Defend will immediately generate a ransomware alert. If *prevent* ransomware is active, {elastic-defend} will terminate the process responsible for modifying the file.

Author: ```suggestion
When ransomware protection is enabled, canary files placed in targeted locations on your hosts provide an early warning system for potential ransomware activity. They provide an early warning system for potential ransomware activity. When canary files are modified, Elastic Defend immediately generates a ransomware alert. If *prevent* ransomware is active, {elastic-defend} terminates the process that modified the file.
```
